### PR TITLE
SIPC Encoding extension

### PR DESCRIPTION
**Pull request by fath11**
This is an extension created by SIPC from Turbowarp ported to Gandi. The extension is under Fath>SIPC Encoded.
Source: https://github.com/TurboWarp/extensions/blob/9394625c0756211449d29845195d950286038516/extensions/encoding.js#L313

### CODE REVIEW

File: example/ccw-approved-ext.js
Comment (sylarhcn) [inline example/ccw-approved-ext.js:1]: 别删除我的 example

**Review by sylarhcn — CHANGES_REQUESTED** [review]
为什么要删除我的 example 啊

File: example/ccw-approved-ext.js
Comment (fath11) [reply example/ccw-approved-ext.js:1]: ok sorry

**Comment by fath11** [comment]
> 为什么要删除我的 example 啊

i added it back, sorry

**Comment by sylarhcn** [comment]
ok，this PR is fine to merge.
the original creator of this extension is "- SPIC -", how do I label this extension author? like "@fath11 & @SPIC"?
and I dont know if SPIC is ok aboout release this extension in Gandi, but I can contact and ask him.
If he doesn't agree, the extension may not be released.

**Comment by sylarhcn** [comment]
good news! I got his permission(in Chinese 😄)，he totaly fine with this PR。

image

File: Fath/SIPC Encoding.js [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+/* eslint-enable */
+
+class SIPC_Encoding {
+  constructor(runtime) {
+    this.runtime = runtime
+  })
+}
+
+formatMessage(id) {
+  return this._formatMessage({
```
Comment (sylarhcn) [inline Fath/SIPC Encoding.js:410]: this._formatMessage maybe undefined?

File: Fath/SIPC Encoding.js [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+  Extension: SIPC_Encoding,
+  info: {
+    name: 'fath.SIPC_Encoding',
+    description: 'fath.description',
+    extensionId: 'Encoding',
+    // iconURL: icon,
+    // insetIconURL: cover,
+    featured: true,
+    disabled: false,
+    collaborator: 'only for hcn test',
```
Comment (sylarhcn) [inline Fath/SIPC Encoding.js:647]: collaborator should be author's name

File: Fath/SIPC Encoding.js (hunk 1 above)
Comment (sylarhcn) [reply Fath/SIPC Encoding.js:410]: _formatMessage is for localization.
if you cant do translate, that's fine, I can help to do that.

File: .replit
Comment (sylarhcn) [inline .replit:1]: suggest ignore replit config file in .gitingore

File: replit.nix
Comment (sylarhcn) [inline replit.nix:1]: suggest ignore replit config file in .gitingore

File: Fath/SIPC Encoding.js (hunk 1 above)
Comment (fath11) [reply Fath/SIPC Encoding.js:410]: yes, thank you

File: Fath/SIPC Encoding.js (hunk 2 above)
Comment (fath11) [reply Fath/SIPC Encoding.js:647]: done

File: .replit
Comment (fath11) [reply .replit:1]: i suppose i just delete it

**Comment by SIPC** [comment]
Don't close it. It's almost on the shelf.

**Comment by fath11** [comment]
> Don't close it. It's almost on the shelf.

i made a new version of the port that I have confirmed is working https://github.com/fath11/custom-extension/blob/main/Fath/SIPC%20Encoding.js

**Pull request closed without merging**